pkgs/by-name/av/avbroot/package.nix

```
@@ -0,0 +1,36 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, gitUpdater
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "avbroot";
+  version = "3.4.0";
+
+  src = fetchFromGitHub {
+    owner = "chenxiaolong";
+    repo = "avbroot";
+    rev = "v${version}";
+    hash = "sha256-RoeanW4vCFm7TCr8T8akNnDI10gqXsMjqq39Y/7LL9o=";
+  };
+
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
+    outputHashes = {
+      "bzip2-0.4.4" = "sha256-9YKPFvaGNdGPn2mLsfX8Dh90vR+X4l3YSrsz0u4d+uQ=";
+      "zip-0.6.6" = "sha256-oZQOW7xlSsb7Tw8lby4LjmySpWty9glcZfzpPuQSSz0=";
+    };
+  };
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
+
+  meta = with lib; {
+    description = "Sign (and root) Android A/B OTAs with custom keys while preserving Android Verified Boot";
+    homepage = "https://github.com/chenxiaolong/avbroot";
+    changelog = "https://github.com/chenxiaolong/avbroot/blob/${src.rev}/CHANGELOG.md";
+    license = with licenses; gpl3;
```

prefer the form:
```suggestion
    license = licenses.gpl3;
```

---

i like this pattern in metadata because it's easier to e.g. add more licenses and adds absolutely no confusion — as `with` often does — since it'll never be more complicated than _at most_ a list. i'd prefer to keep it like this, but i also don't care enough to not have this merged because of this. :)

---

emm, that maybe you want this ...
```suggestion
    license = with licenses; [ gpl3 ];
```

---

By the ways, currently it's not recommand to use with lib in meta anymore.
you may want to remove it with lib. at the front of each of it.